src/common/os_dimm_ndctl.c

```diff
@@ -104,11 +136,27 @@ os_dimm_region_namespace(struct ndctl_ctx *ctx, const os_stat_t *st,
 		} else if ((dax = ndctl_namespace_get_dax(ndns))) {
 			struct daxctl_region *dax_region;
 			dax_region = ndctl_dax_get_daxctl_region(dax);
-			/* there is always one dax device in dax_region */
 			if (dax_region) {
 				struct daxctl_dev *dev;
-				dev = daxctl_dev_get_first(dax_region);
-				devname = daxctl_dev_get_devname(dev);
+				daxctl_dev_foreach(dax_region, dev) {
+					if (!dev)
+						continue;
+					devname = daxctl_dev_get_devname(dev);
+					int ret = os_dimm_match_device(st,
+						devname);
+					if (ret < 0) {
+						return ret;
+					} else if (ret) {
+						*pregion = region;
+						if (pndns)
+							*pndns = ndns;
+
+						return 0;
+					} else {
+						continue;
```

Comment:
you don't need this continue, or it doesn't do what you think it does.
Fix these loops, the control flow is bad.